src/plugins/global-settings/typography/index.js

```diff
@@ -88,23 +159,90 @@ addFilter( 'stackable.global-settings.inspector', 'stackable/global-typography',
+			let typeScale = _typographySettings?.h6?.fontSize
+			if ( typeScale ) {
+				const computedApplied = getAppliedTypeScale( typeScale ) ?? {}
+				const tags = Object.keys( _typographySettings )
+				for ( const tag of tags ) {
+				// If font size mismatch, set typography scale to Custom
+					if ( _typographySettings[ tag ]?.fontSize !== computedApplied[ tag ]?.fontSize ) {
+						typeScale = 'custom'
+					}
+				}
+				setSelectedTypeScale( typeScale )
+			}
 		} )
 	}, [] )
 
-	// When typography styles are changed, trigger our editor style generator to update.
 	useEffect( () => {
-		doAction( 'stackable.global-settings.typography.update-trigger', typographySettings, applySettingsTo )
-	}, [ JSON.stringify( typographySettings ), applySettingsTo ] )
+		// When typography styles are changed, trigger our editor style generator to update.
+		doAction( 'stackable.global-settings.typography.update-trigger', typographySettings, applySettingsTo, isApplyBodyToHTML )
+	}, [ JSON.stringify( typographySettings ), applySettingsTo, isApplyBodyToHTML ] )
+
+	useEffect( () => {
+		// When typography styles are changed, trigger our editor style generator to update.
+		doAction( 'stackable.global-settings.typography.update-trigger', typographySettings, applySettingsTo, isApplyBodyToHTML )
+
+		// Update the custom presets when using typography as presets
+		if ( useTypographyAsPresets ) {
+			const fontSizePresets = TYPOGRAPHY_TAGS
+				.filter( ( { presetSlug } ) => !! presetSlug )
+				.map( ( {
+					selector, presetName, presetSlug,
+				} ) => {
+					const size = typographySettings[ selector ]?.fontSize ?? getDefaultFontSize( selector ) ?? 16
+					const unit = typographySettings[ selector ]?.fontSizeUnit ?? 'px'
+					return {
+						name: presetName,
+						slug: presetSlug,
+						size: `${ size }${ unit }`,
+					}
+				} )
+			// Add the preset for extra small
+			let xSmallSize = typographySettings[ '.stk-subtitle' ]?.fontSize ?? getDefaultFontSize( '.stk-subtitle' ) ?? 16
+			let xSmallUnit = typographySettings[ '.stk-subtitle' ]?.fontSizeUnit ?? 'px'
+			if ( xSmallUnit === 'px' ) {
+				xSmallSize = Math.pow( xSmallSize / 16, 2 )
+				xSmallUnit = 'rem'
+			} else {
+				xSmallSize = Math.pow( xSmallSize, 2 )
+			}
+
+			fontSizePresets.push( {
+				name: 'XS',
+				slug: 'x-small',
+				size: `${ xSmallSize }${ xSmallUnit ?? 'px' }`,
+			} )
+			// Reverse the presets so it's from smallest to biggest
+			fontSizePresets.reverse()
+
+			const newSettings = { ...allCustomPresets, fontSizes: fontSizePresets }
+
+			clearTimeout( saveTypographyAsPresetsThrottle )
+			saveTypographyAsPresetsThrottle = setTimeout( () => {
+				const settings = new models.Settings( { stackable_global_custom_preset_controls: newSettings } ) // eslint-disable-line camelcase
+				settings.save()
+			}, 300 )
+
+			dispatch( 'stackable/global-preset-controls.custom' ).updateCustomPresetControls( newSettings )
```

Comment:
Looks like this code updates the preset controls and belongs to the preset controls area. In the preset controls area, you can just listen to calls to `stackable.global-settings.typography.update-trigger` and perform preset changes accordingly